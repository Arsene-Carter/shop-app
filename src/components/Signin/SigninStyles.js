import { styled } from 'styletron-react';
import { BodyText, Input, Link, Button } from '../../ui';

const inputItemStyles = {
  marginBottom: '10px'
};

export const LoginButton = styled(Button, {
  color: 'white',
  ...inputItemStyles
});

export const RegisterLink = styled(Link, {
  marginLeft: '20px'
});

export const RegisterText = styled(BodyText, {
  textAlign: 'center',
  display: 'flex',
  flexDirection: 'row',
  justifyContent: 'center',
  alignItems: 'center',
});

export const SigninInput = styled(Input, inputItemStyles);



