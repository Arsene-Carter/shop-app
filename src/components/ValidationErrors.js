export const REQUIRED = 'This field is required'
export const INVALID_EMAIL = 'This must be a valid email address'
export const MIN_6_CHARACTERS = 'Password must have at least 6 characters'
export const PASSWORD_NOT_MATCH = 'Password and confirm password not match'
