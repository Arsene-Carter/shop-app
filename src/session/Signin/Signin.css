.SignIn {
    display: flex;
}

.SignIn .topSpace {
    height: 50px;
}