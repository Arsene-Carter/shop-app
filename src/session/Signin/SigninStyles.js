import { styled } from 'styletron-react';

export const SiginContainer = styled('div', {
  width: '30%',
  display: 'flex',
  flexDirection: 'column',
  alignIntems: 'center',
  alignSelf: 'center',
});

export const loginBtn = {
  alignSelf: 'center',
  color: 'white',
  marginTop: '20px',
};

export const inputStyles = {
  marginBottom: '10px',
};

export const RegisterText = styled('p', {
  textAlign: 'center',
});

