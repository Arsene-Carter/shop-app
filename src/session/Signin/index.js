import React, { Component, PropTypes } from 'react';
import { styled } from 'styletron-react';
import { Button } from '../../ui';

export default class SignIn extends Component {
  constructor(props) {
    super(props);
    this.state = {};
    console.log('props', this.props);
  }

  changeHandler = name => (e) => {
    this.setState({
      [name]: e.target.value,
    });
  }

  render() {
    const { email, password } = this.state;
    const { login } = this.props;

    return (
      <div>
        <input
          onChange={this.changeHandler('email')}
        />
        <input
          type="password"
          onChange={this.changeHandler('password')}
        />
        <Button
          onClick={() => login({ email, password }).then(() => {

            this.props.getLists();
          })}
        >
          SignIn
        </Button>
      </div>
    )
  }
}

SignIn.propTypes = {
  login: PropTypes.func.isRequired,
};


