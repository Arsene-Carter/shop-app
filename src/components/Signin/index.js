// @flow
import React, { Component } from 'react';
import { SessionContainer } from '../styles';
import { RegisterText, RegisterLink, LoginButton, SigninTitle } from './SigninStyles';
import { FormField, Input } from '../../ui';
import * as validationErrors from '../ValidationErrors';
import FormWithError from '../FormComponents/FormError';

type SigninFields = {
  email: string,
  password: string
};

type TouchedState = {
  email: boolean,
  password: boolean
};

type State = {
  email: string,
  password: string,
  touchedFields: {
    email: boolean,
    password: boolean
  }
};

type Props = {
  login: ({
    email: string,
    password: string
  }) => void
};

export const validate = ({
  email,
  password,
  touchedFields
}: SigninFields & { touchedFields: TouchedState }): {
  email: ?string,
  password: ?string
} => {
  const validateRequired = (field: string) =>
    field ? '' : validationErrors.REQUIRED;
  const validateEmail = (emailStr: string) => {
    if (
      !emailStr.match(
        /^[a-zA-Z0-9.!#$%&â€™*+/=?^_`{|}~-]+@[a-zA-Z0-9-]+(?:\.[a-zA-Z0-9-]+)*$/
      )
    ) {
      return validationErrors.INVALID_EMAIL;
    }
    return null;
  };

  const minChars = len => str =>
    str.length < len ? validationErrors.MIN_6_CHARACTERS : '';
  const min6 = minChars(6);
  const isTouched = field => touchedFields[field];

  return {
    email:
      (isTouched('email') &&
        (validateRequired(email) || validateEmail(email))) ||
        '',
    password:
      (isTouched('password') &&
        (validateRequired(password) || min6(password))) ||
        ''
  };
};

const WithErrorInput = FormWithError(Input);

export default class SignIn extends Component<void, Props, State> {
  state = {
    email: '',
    password: '',
    touchedFields: {
      email: false,
      password: false
    }
  };

  changeHandler = (e: SyntheticInputEvent) => {
    this.setState({ [e.target.name]: e.target.value });
  };

  loginClickHandler = () => {
    this.setState(
      {
        touchedFields: {
          email: true,
          password: true
        }
      },
      () => {
        const errors = validate(this.state);
        const areErrors = errObj =>
          Object.keys(errObj)
            .map(key => errors[key])
            .some(Boolean);

        if (!areErrors(errors)) {
          this.props.login({
            email: this.state.email,
            password: this.state.password
          });
        }
      }
    );
  };

  render() {
    const errors = validate(this.state);

    return (
      <SessionContainer>
        <SigninTitle center>Sign in to Shopping List</SigninTitle>
        <FormField>
          <WithErrorInput
            error={errors.email}
            placeholder="Enter email"
            name="email"
            onChange={this.changeHandler}
          />
        </FormField>
        <FormField>
          <WithErrorInput
            error={errors.password}
            placeholder="Enter password"
            name="password"
            type="password"
            onChange={this.changeHandler}
          />
        </FormField>

        <LoginButton primary onClick={this.loginClickHandler}>
          Login
        </LoginButton>

        <RegisterText>
          Not a member?
          <RegisterLink to="/register">Register</RegisterLink>
        </RegisterText>
      </SessionContainer>
    );
  }
}
