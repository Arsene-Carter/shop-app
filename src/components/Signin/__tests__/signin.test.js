import React from 'react'
import renderer from 'react-test-renderer';
import { MemoryRouter } from 'react-router-dom';
import WithStyletron from '../../../shared/styletron/configure';
import Signin, { validate } from '../';
import * as validationErrors from '../../ValidationErrors'

test('render signin', () => {
  const signinTree = renderer.create(
    <MemoryRouter>
      <WithStyletron>
        <Signin />
      </WithStyletron>
    </MemoryRouter>
  )

  expect(signinTree).toMatchSnapshot()
})

test('validate check for required fields', () => {
  const errors = validate({
    email: '',
    password: '',
    touchedFields: {
      email: true,
      password: true
    }
  })

  expect(errors).toEqual({
    email: validationErrors.REQUIRED,
    password: validationErrors.REQUIRED
  })
})


test('validate return error for incorrect email', () => {
  const errors = validate({
    email: 'ioan',
    password: '',
    touchedFields: {
      email: true,
      password: true
    }
  })

  expect(errors).toEqual({
    email: validationErrors.INVALID_EMAIL,
    password: validationErrors.REQUIRED
  })
})
