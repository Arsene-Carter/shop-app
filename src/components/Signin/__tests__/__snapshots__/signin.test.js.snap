// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render signin 1`] = `
<div
  className="a b c d e f g h i j k l m n o p q r s t u v"
>
  <h2
    className="w"
  >
    Sign in to Shopping List
  </h2>
  <div
    className={undefined}
  >
    <div>
      <div
        className={undefined}
      >
        <input
          className=""
          name="email"
          onChange={[Function]}
          placeholder="Enter email"
          type="text"
        />
      </div>
      
    </div>
  </div>
  <div
    className={undefined}
  >
    <div>
      <div
        className={undefined}
      >
        <input
          className=""
          name="password"
          onChange={[Function]}
          placeholder="Enter password"
          type="password"
        />
      </div>
      
    </div>
  </div>
  <a
    className="x"
    onClick={[Function]}
  >
    Login
  </a>
  <p
    className="y z a0 a1 a2 a3 a4 b c d e f a5 a6 a7 j a8 a9 aa ab ac ad ae af ag"
  >
    Not a member?
    <a
      className="ah"
      href="/register"
      onClick={[Function]}
    >
      Register
    </a>
  </p>
</div>
`;
